compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -826,6 +826,9 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
         false
       } else isSubType(tp1, tp2, approx.addLow)
 
+    def isSubApproxHi(tp1: Type, tp2: Type): Boolean =
+      (tp2 ne NothingType) && isSubType(tp1, tp2, approx.addHigh)
```

Comment:
Is the Nothing check an optimization? Any chance it could do the wrong thing if `tp1` is a refinement of Nothing like `Nothing { type T }` ?